kv/dist_sender.go

```diff
@@ -467,8 +467,12 @@ func (ds *DistSender) Send(ctx context.Context, ba roachpb.BatchRequest) (*roach
 			// Next, zero out the NodeList pointer. That makes sure that
 			// if we had something of size zero but with capacity, we don't
 			// re-use the existing space (which others may also use).
-			txnShallow.CertainNodes.Nodes = nil
-			txnShallow.CertainNodes.Add(nDesc.NodeID)
+			// This is just to satisfy paranoia/OCD and not expected to matter
+			// in practice.
+			txnShallow.MaxTimestamps = nil
+			// OrigTimestamp is the HLC timestamp at which the Txn started, so
+			// this effectively means no more uncertainty on this node.
+			txnShallow.UpdateUncertainty(nDesc.NodeID, txnShallow.OrigTimestamp)
```

Comment:
Why not call this `UpdateMaxTimestamps`? The conflation of uncertainty intervals concept here seems unnecessarily confusing.